<a href="https://colab.research.google.com/github/rakshansingh12/disease-risk-prediction-ml/blob/main/02_data_loading_and_exploration.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import os

PROJECT_PATH = "/content/drive/MyDrive/disease-risk-prediction-gene-expression"
RAW_DATA_PATH = os.path.join(PROJECT_PATH, "data/raw")

os.listdir(RAW_DATA_PATH)

['GSE5281_expression_matrix.txt.gz', 'GPL570_annotation.txt']

In [3]:
import pandas as pd

file_path = os.path.join(RAW_DATA_PATH, "GSE5281_expression_matrix.txt.gz")

data = pd.read_csv(
    file_path,
    sep="\t",
    compression="gzip",
    comment="!",
    index_col=0
)

In [4]:
data.shape

(54675, 161)

In [6]:
X_full = data.T

In [ ]:
data.head()

GSM119615   GSM119616    GSM119617    GSM119618   GSM119619  \
ID_REF                                                                    
1007_s_at  693.973300  807.145140  1722.625000  1668.057500  2231.29690   
1053_at      9.755318   19.052270    13.060389    14.652245   138.67683   
117_at      13.346846   18.217276    97.425470    98.197945   106.46844   
121_at     173.092450  289.315400   537.182800   496.346560   477.86517   
1255_g_at   52.616950  110.375984    99.021736    17.306772   628.81616   

            GSM119620   GSM119621    GSM119622   GSM119623   GSM119624  ...  \
ID_REF                                                                  ...   
1007_s_at  990.541000  1396.82420  1026.056500  736.770300  938.817100  ...   
1053_at      7.186742    10.17089    34.056934    7.829473   40.235332  ...   
117_at      98.710520   164.32666    46.306164   35.067910   72.864900  ...   
121_at     355.092000   467.56790   294.583280  224.524060  298.284730  ...   
1255_g_at  105.254600   142.26918   123.016136  183.396910  180.814250  ...   

           GSM238945  GSM238946  GSM238947   GSM238948  GSM238949  GSM238951  \
ID_REF                                                                         
1007_s_at   983.1340   855.5980  2074.2300  1385.89000  1390.9400   904.4160   
1053_at      78.0901    57.5839     4.9474    11.06460    78.2078    50.5737   
117_at       35.8215   106.8380    87.6188    39.50480   117.6100    20.6184   
121_at      114.5470   174.5280   180.7250   238.92100   221.8730   153.4310   
1255_g_at    18.8389    80.0143    69.0884     5.82269    44.7857    11.4650   

           GSM238952  GSM238953   GSM238955  GSM238963  
ID_REF                                                  
1007_s_at  1319.0900   783.2270  1552.96000  2845.4500  
1053_at      13.7651    56.3892     6.97840    33.0147  
117_at       31.4041    41.2217    40.48350    19.1316  
121_at      262.5170   136.9680   284.89900   404.0750  
1255_g_at    30.3939    40.7314     4.34157    42.2666  

[5 rows x 161 columns]

In [7]:
X_full.shape

(161, 54675)

In [8]:
sample = X_full.iloc[0].values

In [9]:
sample.shape

(54675,)

In [10]:
import numpy as np

np.savetxt(
    "example_sample.csv",
    sample.reshape(1, -1),
    delimiter=","
)

In [ ]:
data.index[:5]

Index(['1007_s_at', '1053_at', '117_at', '121_at', '1255_g_at'], dtype='object', name='ID_REF')

In [ ]:
data.columns[:5]

Index(['GSM119615', 'GSM119616', 'GSM119617', 'GSM119618', 'GSM119619'], dtype='object')

In [ ]:
X = data.T
X.shape

(161, 54675)

We have gene expression measurements for 54,675 (columns) genes across 161 (rows) human samples.